File: EditLinks.ascx.vb
'
' DotNetNuke® - http://www.dotnetnuke.com
' Copyright (c) 2002-2008
' by DotNetNuke Corporation
'
' Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
' documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
' the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
' to permit persons to whom the Software is furnished to do so, subject to the following conditions:
'
' The above copyright notice and this permission notice shall be included in all copies or substantial portions 
' of the Software.
'
' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
' TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
' THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
' CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
' DEALINGS IN THE SOFTWARE.
'

Imports System.Web.UI.WebControls
Imports DotNetNuke
Imports DotNetNuke.Entities.Modules
Imports DotNetNuke.UI.Utilities

Namespace DotNetNuke.Modules.Links

    ''' -----------------------------------------------------------------------------
    ''' <summary>
	''' The EditLinks PortalModuleBase is used to manage the Links
	''' </summary>
    ''' <remarks>
	''' </remarks>
	''' <history>
	''' 	[cnurse]	9/23/2004	Moved Links to a separate Project
	''' 	[cnurse]	9/23/2004	Updated to reflect design changes for Help, 508 support
	'''                       and localisation
	''' </history>
	''' -----------------------------------------------------------------------------
	Partial  Class EditLinks
		Inherits Entities.Modules.PortalModuleBase

#Region " Private Members "

        Private itemId As Integer = -1

#End Region

#Region " Event Handlers "


        ''' -----------------------------------------------------------------------------
        ''' <summary>
        ''' Page_Load runs when the control is loaded
        ''' </summary>
        ''' <remarks>
        ''' </remarks>
        ''' <history>
        ''' 	[cnurse]	9/23/2004	Updated to reflect design changes for Help, 508 support
        '''                       and localisation
        ''' </history>
        ''' -----------------------------------------------------------------------------
        Private Sub Page_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

            Try

                Dim objModules As New Entities.Modules.ModuleController
                Dim objLinks As New LinkController
                Dim roleController As New DotNetNuke.Security.Roles.RoleController
                ' Determine ItemId of Link to Update

                If Not (Request.QueryString("ItemId") Is Nothing) Then
                    itemId = Int32.Parse(Request.QueryString("ItemId"))
                End If

                ' If the page is being requested the first time, determine if an
                ' link itemId value is specified, and if so populate page
                ' contents with the link details
                If Page.IsPostBack = False Then

                    tblGetContent.Visible = System.Security.SecurityManager.IsGranted(New System.Net.WebPermission(System.Net.NetworkAccess.Connect, "http://www.dotnetnuke.com"))

                    Dim allUsersRole As New DotNetNuke.Security.Roles.RoleInfo()
                    allUsersRole.RoleID = -2
                    allUsersRole.RoleName = "All Users"
                    Dim unAuthorizedUsersRole As New DotNetNuke.Security.Roles.RoleInfo()
                    unAuthorizedUsersRole.RoleID = -1
                    unAuthorizedUsersRole.RoleName = "Unauthenticated Users"

                    Dim roles = roleController.GetPortalRoles(Me.PortalId)
                    roles.Add(unAuthorizedUsersRole)
                    roles.Add(allUsersRole)

                    ddlViewOrderLinks.DataSource = objLinks.GetLinks(Me.ModuleId)
                    ddlViewOrderLinks.DataTextField = "Title"
                    ddlViewOrderLinks.DataValueField = "ViewOrder"
                    ddlViewOrderLinks.DataBind()

                    cblGrantRoles.DataSource = roles
                    cblGrantRoles.DataTextField = "RoleName"
                    cblGrantRoles.DataValueField = "RoleID"
                    cblGrantRoles.DataBind()

                    ClientAPI.AddButtonConfirm(cmdDelete, Services.Localization.Localization.GetString("DeleteItem"))

                    If itemId <> -1 Then

                        ' Obtain a single row of link information

                        Dim objLink As LinkInfo = objLinks.GetLink(itemId, ModuleId)

                        If Not objLink Is Nothing Then

                            ddlViewOrderLinks.Items.Remove(ddlViewOrderLinks.Items.FindByText(objLink.Title))

                            If ddlViewOrderLinks.Items.Count > 0 Then

                                ddlViewOrderLinks.SelectedValue = objLinks.GetLinkByHigherViewOrder(objLink.ViewOrder, Me.ModuleId).ToString

                                If Integer.Parse(ddlViewOrderLinks.SelectedValue) < objLink.ViewOrder Then
                                    ddlViewOrder.SelectedValue = "A"
                                Else
                                    ddlViewOrder.SelectedValue = "B"
                                End If

                            End If

                            txtTitle.Text = objLink.Title.ToString

                            ctlURL.Url = objLink.Url
                            ctlURL.ShowDatabase = True
                            ctlURL.ShowSecure = True
                            'chkGetContent.Checked = objLink.RefreshContent

                            Dim urlType As String = LinkController.convertURLType(GetURLType(objLink.Url))

                            If urlType <> "U" Then
                                tblGetContent.Visible = False
                            End If

                            txtDescription.Text = objLink.Description.ToString

                            'If (Common.Utilities.Null.IsNull(objLink.ViewOrder) = False) Then
                            '    txtViewOrder.Text = Convert.ToString(objLink.ViewOrder)
                            'End If

                            ddlGetContentInterval.SelectedValue = objLink.RefreshInterval

                            ctlAudit.CreatedByUser = objLink.CreatedByUser.ToString
                            ctlAudit.CreatedDate = objLink.CreatedDate.ToString

                            ctlTracking.URL = objLink.Url
                            ctlTracking.ModuleID = ModuleId

                            For Each cb As ListItem In cblGrantRoles.Items
                                cb.Selected = objLink.GrantRoles.Contains(";" & cb.Value & ";")
                            Next

                        Else       ' security violation attempt to access item not related to this Module

                            Response.Redirect(NavigateURL(), True)

                        End If

                    Else

                        cmdDelete.Visible = False
                        ctlAudit.Visible = False
                        ctlTracking.Visible = False

                    End If

                End If

            Catch exc As Exception

                ProcessModuleLoadException(Me, exc)

            End Try

        End Sub

        ''' <summary>
        ''' Hide get content functionality if resource url type is none external url
        ''' </summary>
        ''' <param name="sender"></param>
        ''' <param name="e"></param>
        ''' <remarks></remarks>
        Protected Sub Page_PreRender(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.PreRender

            ' hide get content functionality when externel url isn`t selected
            If Not String.IsNullOrEmpty(ctlURL.UrlType) AndAlso ctlURL.UrlType <> "U" Then tblGetContent.Visible = False

        End Sub

        ''' -----------------------------------------------------------------------------
        ''' <summary>
        ''' cmdCancel_Click runs when the cancel button is clicked
        ''' </summary>
        ''' <remarks>
        ''' </remarks>
        ''' <history>
        ''' 	[cnurse]	9/23/2004	Updated to reflect design changes for Help, 508 support
        '''                       and localisation
        ''' </history>
        ''' -----------------------------------------------------------------------------
        Private Sub cmdCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles cmdCancel.Click

            Try

                Response.Redirect(NavigateURL(), True)

            Catch exc As Exception           'Module failed to load

                ProcessModuleLoadException(Me, exc)

            End Try

        End Sub

        ''' -----------------------------------------------------------------------------
        ''' <summary>
        ''' cmdDelete_Click runs when the delete button is clicked
        ''' </summary>
        ''' <remarks>
        ''' </remarks>
        ''' <history>
        ''' 	[cnurse]	9/23/2004	Updated to reflect design changes for Help, 508 support
        '''                       and localisation
        ''' </history>
        ''' -----------------------------------------------------------------------------
        Private Sub cmdDelete_Click(ByVal sender As Object, ByVal e As EventArgs) Handles cmdDelete.Click

            Try

                If itemId <> -1 Then

                    Dim links As New LinkController
                    links.DeleteLink(itemId, ModuleId)
                    ModuleController.SynchronizeModule(ModuleId)

                End If

                ' Redirect back to the portal home page
                Response.Redirect(NavigateURL(), True)

            Catch exc As Exception

                ProcessModuleLoadException(Me, exc)

            End Try

        End Sub

        ''' -----------------------------------------------------------------------------
        ''' <summary>
        ''' cmdUpdate_Click runs when the update button is clicked
        ''' </summary>
        ''' <remarks>
        ''' </remarks>
        ''' <history>
        ''' 	[cnurse]	9/23/2004	Updated to reflect design changes for Help, 508 support
        '''                       and localisation
        ''' </history>
        ''' -----------------------------------------------------------------------------
        Private Sub cmdUpdate_Click(ByVal sender As Object, ByVal e As EventArgs) Handles cmdUpdate.Click

            Try

                If Page.IsValid = True And ctlURL.Url <> "" Then

                    Dim objLink As New LinkInfo
                    Dim objLinks As New LinkController

                    objLink = CType(CBO.InitializeObject(objLink, GetType(LinkInfo)), LinkInfo)
                    'bind text values to object
                    objLink.ItemId = itemId
                    objLink.ModuleId = ModuleId
                    objLink.CreatedByUser = UserInfo.UserID
                    objLink.CreatedDate = DateTime.Now()
                    objLink.Title = txtTitle.Text
                    objLink.Url = ctlURL.Url

                    Dim refreshInterval As Integer = 0

                    If ctlURL.UrlType = "U" Then
                        refreshInterval = CInt(ddlGetContentInterval.SelectedValue)
                    End If

                    objLink.RefreshInterval = refreshInterval

                    If (ddlViewOrderLinks.Items.Count > 0) Then

                        Select Case ddlViewOrder.SelectedValue
                            Case "B"
                                objLink.ViewOrder = Convert.ToInt32(ddlViewOrderLinks.SelectedValue) - 1
                                objLinks.UpdateViewOrder(objLink, -1, Me.ModuleId)
                            Case "A"
                                objLink.ViewOrder = Convert.ToInt32(ddlViewOrderLinks.SelectedValue) + 1
                                objLinks.UpdateViewOrder(objLink, 1, Me.ModuleId)
                            Case Else
                                objLink.ViewOrder = Null.NullInteger
                        End Select

                    Else
                        objLink.ViewOrder = Null.NullInteger
                    End If

                    objLink.Description = txtDescription.Text
                    objLink.GrantRoles = ";"

                    For Each cb As ListItem In cblGrantRoles.Items
                        If cb.Selected Then
                            objLink.GrantRoles += cb.Value & ";"
                        End If
                    Next

                    If objLink.GrantRoles.Equals(";") Then
                        objLink.GrantRoles += "0;"
                    End If

                    ' Create an instance of the Link DB component

                    If Common.Utilities.Null.IsNull(itemId) Then
                        objLinks.AddLink(objLink)
                    Else
                        objLinks.UpdateLink(objLink)
                    End If

                    ModuleController.SynchronizeModule(ModuleId)

                    ' url tracking
                    Dim objUrls As New UrlController
                    objUrls.UpdateUrl(PortalId, ctlURL.Url, ctlURL.UrlType, ctlURL.Log, ctlURL.Track, ModuleId, ctlURL.NewWindow)

                    ' Redirect back to the portal home page
                    Response.Redirect(NavigateURL(), True)

                End If

            Catch exc As Exception

                ProcessModuleLoadException(Me, exc)

            End Try

        End Sub

        ''' <summary>
        ''' Try to retrieve meta description and title from the url specified
        ''' </summary>
        ''' <param name="sender"></param>
        ''' <param name="e"></param>
        ''' <remarks></remarks>
        ''' <history>
        '''     [alex]      10/02/2009   First implementation 
        ''' </history>
        Protected Sub lbtGetContent_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles lbtGetContent.Click

            Dim targetUrl As String = Me.ctlURL.Url

            If Not String.IsNullOrEmpty(targetUrl) Then

                Dim retrieveMessage As String = Localization.GetString("msgGetContentSucceeded.Text", Me.LocalResourceFile)
                Dim retrieveMessageCssClass As String = "MessageSuccees"

                Try
                    ' get content from target url
                    Dim targetInfo As TargetInfo = New LinkController().GetTargetContent(targetUrl)

                    Me.txtTitle.Text = targetInfo.Title
                    Me.txtDescription.Text = targetInfo.Description

                Catch ex As System.Net.WebException

                    retrieveMessage = Localization.GetString("msgGetContentFailed.Text", Me.LocalResourceFile)
                    retrieveMessageCssClass = "MessageFailure"

                End Try

                lblGetContentResult.Text = retrieveMessage
                lblGetContentResult.CssClass = retrieveMessageCssClass

                valTitle.Validate()

            End If

        End Sub

#End Region

    End Class

End Namespace

File: EditLinks.ascx.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information

namespace DotNetNuke.Modules.Links
{
    using System;
    using System.Security;
    using System.Web.UI.WebControls;

    using DotNetNuke.Common.Utilities;
    using DotNetNuke.Entities.Modules;
    using DotNetNuke.Modules.Links.Components;
    using DotNetNuke.Security.Roles;
    using DotNetNuke.Services.Exceptions;
    using DotNetNuke.Services.Localization;
    using DotNetNuke.UI.Utilities;

    /// <summary>
    /// Codebehind for editing links user interface.
    /// </summary>
    public partial class EditLinks : PortalModuleBase
    {
#pragma warning disable SA1401 // Fields should be private
        /// <summary>
        /// The URL control.
        /// </summary>
        protected UI.UserControls.UrlControl ctlURL;

        /// <summary>
        /// The audit control.
        /// </summary>
        protected UI.UserControls.ModuleAuditControl ctlAudit;

        /// <summary>
        /// The urlTracking control.
        /// </summary>
        protected UI.UserControls.URLTrackingControl ctlTracking;
#pragma warning restore SA1401 // Fields should be private

        private int itemId = -1;

        /// <summary>
        /// Handles the Click event of the cmdCancel control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
#pragma warning disable SA1300 // Element should begin with upper-case letter
        public void cmdCancel_Click(object sender, EventArgs e)
#pragma warning restore SA1300 // Element should begin with upper-case letter
        {
            try
            {
                this.Response.Redirect(DotNetNuke.Common.Globals.NavigateURL(), true);
            }
            catch (Exception exc)
            {
                Exceptions.ProcessModuleLoadException(this, exc);
            }
        }

        /// <summary>
        /// Handles the Click event of the cmdDelete control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
#pragma warning disable SA1300 // Element should begin with upper-case letter
        public void cmdDelete_Click(object sender, EventArgs e)
#pragma warning restore SA1300 // Element should begin with upper-case letter
        {
            try
            {
                if (this.itemId != -1)
                {
                    LinkController.DeleteLink(this.itemId, this.ModuleId);
                    ModuleController.SynchronizeModule(this.ModuleId);
                }

                // Redirect back to the portal home page
                this.Response.Redirect(DotNetNuke.Common.Globals.NavigateURL(), true);
            }
            catch (Exception exc)
            {
                Exceptions.ProcessModuleLoadException(this, exc);
            }
        }

        /// <summary>
        /// Handles the Click event of the cmdUpdate control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
#pragma warning disable SA1300 // Element should begin with upper-case letter
        public void cmdUpdate_Click(object sender, EventArgs e)
#pragma warning restore SA1300 // Element should begin with upper-case letter
        {
            try
            {
                if (this.Page.IsValid == true & this.ctlURL.Url != string.Empty)
                {
                    Link objLink = new Link();

                    // bind text values to object
                    objLink.ItemId = this.itemId;
                    objLink.ModuleId = this.ModuleId;
                    objLink.CreatedByUser = this.UserInfo.UserID;
                    objLink.CreatedDate = DateTime.Now;
                    objLink.Title = this.txtTitle.Text;
                    objLink.Url = this.ctlURL.Url;

                    int refreshInterval = 0;

                    if (this.ctlURL.UrlType == "U")
                    {
                        refreshInterval = System.Convert.ToInt32(this.ddlGetContentInterval.SelectedValue);
                    }

                    objLink.RefreshInterval = refreshInterval;

                    if (this.ddlViewOrderLinks.Items.Count > 0)
                    {
                        switch (this.ddlViewOrder.SelectedValue)
                        {
                            case "B":
                                {
                                    objLink.ViewOrder = Convert.ToInt32(this.ddlViewOrderLinks.SelectedValue) - 1;
                                    LinkController.UpdateViewOrder(objLink, -1, this.ModuleId);
                                    break;
                                }

                            case "A":
                                {
                                    objLink.ViewOrder = Convert.ToInt32(this.ddlViewOrderLinks.SelectedValue) + 1;
                                    LinkController.UpdateViewOrder(objLink, 1, this.ModuleId);
                                    break;
                                }

                            default:
                                {
                                    objLink.ViewOrder = Null.NullInteger;
                                    break;
                                }
                        }
                    }
                    else
                    {
                        objLink.ViewOrder = Null.NullInteger;
                    }

                    objLink.Description = this.txtDescription.Text;
                    objLink.GrantRoles = ";";

                    foreach (ListItem cb in this.cblGrantRoles.Items)
                    {
                        if (cb.Selected)
                        {
                            objLink.GrantRoles += cb.Value + ";";
                        }
                    }

                    if (objLink.GrantRoles.Equals(";"))
                    {
                        objLink.GrantRoles += "0;";
                    }

                    // Create an instance of the Link DB component
                    if (Common.Utilities.Null.IsNull(this.itemId))
                    {
                        LinkController.AddLink(objLink);
                    }
                    else
                    {
                        LinkController.UpdateLink(objLink);
                    }

                    ModuleController.SynchronizeModule(this.ModuleId);

                    // url tracking
                    UrlController objUrls = new UrlController();
                    objUrls.UpdateUrl(this.PortalId, this.ctlURL.Url, this.ctlURL.UrlType, this.ctlURL.Log, this.ctlURL.Track, this.ModuleId, this.ctlURL.NewWindow);

                    // Redirect back to the portal home page
                    this.Response.Redirect(DotNetNuke.Common.Globals.NavigateURL(), true);
                }
            }
            catch (Exception exc)
            {
                Exceptions.ProcessModuleLoadException(this, exc);
            }
        }

        /// <inheritdoc/>
        protected override void OnInit(EventArgs e)
        {
            base.OnInit(e);
            this.Load += this.Page_Load;
        }

        /// <summary>
        /// Handles the Load event of the Page control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        protected void Page_Load(object sender, EventArgs e)
        {
            try
            {
                // Determine ItemId of Link to Update
                if (this.Request.QueryString["ItemId"] != null)
                {
                    int.TryParse(this.Request.QueryString["ItemId"], out this.itemId);
                }

                // If the page is being requested the first time, determine if an
                // link itemId value is specified, and if so populate page
                // contents with the link details
                if (this.Page.IsPostBack == false)
                {
                    var permissionSet = new PermissionSet(System.Security.Permissions.PermissionState.None);
                    permissionSet.AddPermission(new System.Net.WebPermission(System.Net.NetworkAccess.Connect, "http://www.dotnetnuke.com"));
                    this.tblGetContent.Visible = permissionSet.IsSubsetOf(AppDomain.CurrentDomain.PermissionSet);

                    var roles = RoleController.Instance.GetRoles(this.PortalId);

                    this.ddlViewOrderLinks.DataSource = LinkController.GetLinks(this.ModuleId);
                    this.ddlViewOrderLinks.DataTextField = "Title";
                    this.ddlViewOrderLinks.DataValueField = "ViewOrder";
                    this.ddlViewOrderLinks.DataBind();

                    this.cblGrantRoles.DataSource = roles;
                    this.cblGrantRoles.DataTextField = "RoleName";
                    this.cblGrantRoles.DataValueField = "RoleID";
                    this.cblGrantRoles.DataBind();

                    ClientAPI.AddButtonConfirm(this.cmdDelete, Services.Localization.Localization.GetString("DeleteItem"));

                    if (this.itemId != -1)
                    {
                        // Obtain a single row of link information
                        var objLink = LinkController.GetLink(this.itemId, this.ModuleId);

                        if (objLink != null)
                        {
                            this.ddlViewOrderLinks.Items.Remove(this.ddlViewOrderLinks.Items.FindByText(objLink.Title));

                            if (this.ddlViewOrderLinks.Items.Count > 0)
                            {
                                this.ddlViewOrderLinks.SelectedValue = LinkController.GetLinkByHigherViewOrder(objLink.ViewOrder, this.ModuleId).ToString();

                                if (int.Parse(this.ddlViewOrderLinks.SelectedValue) < objLink.ViewOrder)
                                {
                                    this.ddlViewOrder.SelectedValue = "A";
                                }
                                else
                                {
                                    this.ddlViewOrder.SelectedValue = "B";
                                }
                            }

                            this.txtTitle.Text = objLink.Title.ToString();
                            this.ctlURL.Url = objLink.Url;
                            string urlType = LinkController.ConvertUrlType(DotNetNuke.Common.Globals.GetURLType(objLink.Url));

                            if (urlType != "U")
                            {
                                this.tblGetContent.Visible = false;
                            }

                            this.txtDescription.Text = objLink.Description.ToString();
                            this.ddlGetContentInterval.SelectedValue = objLink.RefreshInterval.ToString();

                            this.ctlAudit.CreatedByUser = objLink.CreatedByUser.ToString();
                            this.ctlAudit.CreatedDate = objLink.CreatedDate.ToString();

                            this.ctlTracking.URL = objLink.Url;
                            this.ctlTracking.ModuleID = this.ModuleId;

                            foreach (ListItem cb in this.cblGrantRoles.Items)
                            {
                                cb.Selected = objLink.GrantRoles.Contains(";" + cb.Value + ";");
                            }
                        }
                        else
                        {
                            this.Response.Redirect(Common.Globals.NavigateURL(), true);
                        }
                    }
                    else
                    {
                        this.cmdDelete.Visible = false;
                        this.ctlAudit.Visible = false;
                        this.ctlTracking.Visible = false;
                    }
                }
            }
            catch (Exception exc)
            {
                Exceptions.ProcessModuleLoadException(this, exc);
            }
        }

        /// <summary>
        /// Handles the PreRender event of the Page control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        protected void Page_PreRender(object sender, System.EventArgs e)
        {
            // hide get content functionality when externel url isn`t selected
            if (!string.IsNullOrEmpty(this.ctlURL.UrlType) && this.ctlURL.UrlType != "U")
            {
                this.tblGetContent.Visible = false;
            }
        }

        /// <summary>
        /// Handles the Click event of the lbtGetContent control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
#pragma warning disable SA1300 // Element should begin with upper-case letter
        protected void lbtGetContent_Click(object sender, EventArgs e)
#pragma warning restore SA1300 // Element should begin with upper-case letter
        {
            string targetUrl = this.ctlURL.Url;

            if (!string.IsNullOrEmpty(targetUrl))
            {
                string retrieveMessage = Localization.GetString("msgGetContentSucceeded.Text", this.LocalResourceFile);
                string retrieveMessageCssClass = "MessageSuccees";

                try
                {
                    // get content from target url
                    TargetInfo targetInfo = LinkController.GetTargetContent(targetUrl);

                    this.txtTitle.Text = targetInfo.Title;
                    this.txtDescription.Text = targetInfo.Description;
                }
                catch (System.Net.WebException)
                {
                    retrieveMessage = Localization.GetString("msgGetContentFailed.Text", this.LocalResourceFile);
                    retrieveMessageCssClass = "MessageFailure";
                }

                this.lblGetContentResult.Text = retrieveMessage;
                this.lblGetContentResult.CssClass = retrieveMessageCssClass;

                this.valTitle.Validate();
            }
        }
    }
}
